Adding new English Paraphraser that retains entities (#368)
cisco/mindmeld: source/userguide/augmentation.rst

``'retain_entities'`` (:class:`bool`, default: ``False``): Whether entity values and annotations should be retained. This is only applicable for the :class:`EnglishParaphraser`.

``'paths'`` (:class:`list`): List of regex based path rules to select files to be augmented. These files become the baseline for the augmentors and are used to generate new data. By default, all training and testing files are augmented.

``'path_suffix'`` (:class:`str`): The default suffix that is appended to the name of the original file being augmented to generate new files with the augmented data.

English Paraphraser
-------------------

The English paraphraser uses a state-of-the-art text summarization model `PEGASUS <https://ai.googleblog.com/2020/06/pegasus-state-of-art-model-for.html>`_ for generating paraphrases. This model has been fine-tuned for the task of paraphrasing.

Usage
^^^^^

.. code-block:: console

    mindmeld augment --app_path <app_path> --language "en"

In the config for this paraphraser class, the ``'augmentor_class'`` should be set to :class:`EnglishParaphraser`.

Retaining Entity Annotations
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> The default English model ignores entity annotations and paraphrases the full query (along with entities). If you wish to retain the entity values and annotations in your query as is in the paraphrases, you can set the ``'retain_entities'`` key to ``True`` in the config.

Reviewer: ```suggestion
The default English model ignores entity annotations and paraphrases the full query (along with entities). If you wish to retain the original query's entity values and annotations in the paraphrases, you can set the ``'retain_entities'`` key to ``True`` in the config.
```

> In addition, if a gazetteer is provided for the entity type, the system will automatically replace the entity values with randomly sampled entities from the gazetteer. This is recommended to improve the n-gram diversity in your queries.

Reviewer: ```suggestion
In addition, if a gazetteer is provided for an entity type, the system will automatically replace entity values with randomly sampled entities from this gazetteer. This is recommended to improve the n-gram diversity in your queries.
```
